### PR TITLE
Deprecate `#boot` and `#boot!` methods

### PR DESCRIPTION
**Pull request by davydovanton**
Reference to #51

### CODE REVIEW

**Comment by solnic** [comment]
Thanks for opening this PR, unfortunately it's not everything because there's no `init` and `start` methods yet :)

**Comment by davydovanton** [comment]
@solnic 😅
So, if you say what exactly you (or other folks) want to see in this methods I can try to update this PR 👍

**Comment by GustavoCaso** [comment]
@davydovanton I think what @solnic means is change the name of the method `boot` with `init` and `boot!` with `start`

**Comment by solnic** [comment]
Yes, exactly what @GustavoCaso said. deprecation API supports creating deprecated methods dynamically so it should be straight-forward. Methods in `Booter` can be simply renamed without deprecating anything.

**Comment by davydovanton** [comment]
okay, I'll update PR :)
Thanks for help @solnic @GustavoCaso 👏

**Comment by GustavoCaso** [comment]
@davydovanton are still interested in updating the PR?

**Comment by davydovanton** [comment]
@GustavoCaso hey! Sorry for delay, I'll update PR today :)

**Comment by solnic** [comment]
@davydovanton no worries man, there are no delays in OSS, things are done when they are done :)

**Comment by GustavoCaso** [comment]
:)

**Comment by davydovanton** [comment]
Done, thanks again for help and review 👏

File: lib/dry/system/container.rb [hunk 1]
```diff
@@ -260,37 +260,37 @@ def finalize!(&block)
         # As a result, `init` and `start` lifecycle triggers are called
         #
         # @example
-        #   MyApp.boot!(:persistence)
+        #   MyApp.init(:persistence)
```
Comment (AMHOL) [inline lib/dry/system/container.rb]: Think this should be `MyApp.start`?

Comment (davydovanton) [inline lib/dry/system/container.rb]: 😅

**Comment by davydovanton** [comment]
folks, any updates? :)

**Comment by timriley** [comment]
@davydovanton Thanks for the reminder, I'll get this merged in today :)

**Comment by timriley** [comment]
Thanks @davydovanton!

**Comment by davydovanton** [comment]
Thanks for the review and merge! 💚

**Pull request merged**